### DISCUSSION TITLE
Better support for PlasmaPro™ 400S?

### DISCUSSION THREAD

**Discussion by dfbills** [RaresAil/homebridge-levoit-air-purifier#30] [Ideas]
Thanks for creating this plugin.  I just connected it to my Levoit PlasmaPro™ 400S Smart True HEPA Air Purifier and was surprised to see how little control is exposed via this plugin. 

I'm only seeing Purifier power state and fan percentage in the first tile and a subjective air quality rating in the second tile.

<img width="332" alt="Screen Shot 2022-04-02 at 1 27 07 PM" src="https://user-images.githubusercontent.com/238797/161394303-f681e589-a61b-424f-8ccc-5fd0039897e5.png">

I was expecting to see a more complete set of controls with near parity to the VeSync app.  Missing:

1. Ability to power toggle
2. Display accessory as Fan type with speed readout 
3. Fan speed control and auto fan mode
4. Sleep mode
5. Plasma Pro control
6. Display on/off

And for a stretch #6, Auto Model Settings control.  Any chance we could work together on exposing these controls?

**Comment by RaresAil** [comment]
Hello, what application do you use, is this the Homebridge Dashboard? From the home app from Apple, you have all these features except the Plasma Pro control and display on/off

**Comment by dfbills** [comment]
Hey @RaresAil- thanks for getting back to me.  I was initially looking at the Homebridge dashboard accessories screen.  After diving deeper into the HomeKit panels, I do indeed see them.  I was expecting them to be exposed more forward like on the Levoit Humidifers- see below:

![Screenshot 2022-04-02 at 7 16 04 PM](https://user-images.githubusercontent.com/238797/161404505-80294d88-bb8f-4ca3-a8fe-e96f0a3f74ba.png)

But this definitely works.  Any chance of getting the Plasma controls into the characteristics?

**Reply by RaresAil** [reply]
I don't own a series with Plasma feature, maybe in future i will work on that feature. The reason on why for humidifier you can see all of them there, is because that plugin exposes all the features as services and in my plugin i expose the features as characteristics for the main service

**Reply by RaresAil** [reply]
I will research on plasma api to see how i can control it, but if you can run in debug mode and share the logs to see how is set in the rersponses

**Reply by dfbills** [reply]
Happy to debug- can you give me a bit more direction on how to do this?  Are you talking about HomeBridge or like Charles?

**Comment by RaresAil** [comment]
@dfbills Sorry for the late reply, please add the following to your config file in homebridge: 
https://github.com/RaresAil/homebridge-levoit-air-purifier#enabling-debug-mode

And then restart the plugin, if it is a child bridge (which is recommended to be), if is not a child bridge, restart the entire homebridge

**Comment by RaresAil** [comment]
As there was no responnse i marked it as no response
